.github/workflows/_rust.yml

```diff
@@ -68,6 +68,52 @@ jobs:
       - uses: ./.github/actions/setup-rust
       - run: cargo test --all-features ${{ matrix.packages }}
 
+  # TODO: Naming
+  # Runs the Windows client smoke test, built in debug mode. We can't run it in release
+  # mode because of a known issue: <https://github.com/firezone/firezone/blob/456e044f882c2bb314e19cc44c0d19c5ad817b7c/rust/windows-client/src-tauri/src/client.rs#L162-L164>
+  windows-smoke-test:
+    name: windows-smoke-test
+    runs-on: windows-2019
+    defaults:
+      run:
+        working-directory: ./rust/windows-client
+    strategy:
+      fail-fast: false
+    env:
+      RUST_LOG: firezone_windows_client=info
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/setup-rust
+        with:
+          targets: x86_64-pc-windows-msvc
+      - name: Run smoke test
+        run: |
+          # Run smoke test
+
+          # Install WebView2
+
+          # This is the "Evergreen" bootstrapper from Microsoft
+          # <https://developer.microsoft.com/en-us/microsoft-edge/webview2/?form=MA13LH#download>
+          # This makes the test non-deterministic, but controlling the version is difficult.
+          Invoke-WebRequest -Uri https://go.microsoft.com/fwlink/p/?LinkId=2124703 -OutFile WebView2Installer.exe
+
+          # This downloads about 200 MB and takes about 5 minutes on my VM
+          # So we could fault in WebView2 without the MSI if we needed.
+          .\WebView2Installer.exe /install
+
+          # Copied from `build.bat`, but I don't want to run `tauri build`
+          pnpm install
+          cp "node_modules/flowbite/dist/flowbite.min.js" "src/"
+          pnpm tsc
+          pnpm tailwindcss -i src/input.css -o src/output.css
+
+          cargo run -p firezone-windows-client -- smoke-test
+          if (-not $?) {
+            echo "Smoke test failed."
+            exit 1
+          }
+
```

Comment:
I'm surprised the Evergreen runtime [isn't preinstalled on the Windows runners](https://github.com/actions/runner-images/blob/main/images/windows/Windows2019-Readme.md). But I suppose we may be able to cache this with a little bit of refactoring.

My first suggestion would be to split up the individual steps so we can more easily wrap the slow steps with https://github.com/actions/cache.

I'll do a bit more poking this morning to see about adding the cache, but splitting this up might be a good start:

```suggestion
      - name: Download WebView2 installer
        run: |
          # This is the "Evergreen" bootstrapper from Microsoft
          # <https://developer.microsoft.com/en-us/microsoft-edge/webview2/?form=MA13LH#download>
          # This makes the test non-deterministic, but controlling the version is difficult.
          Invoke-WebRequest -Uri https://go.microsoft.com/fwlink/p/?LinkId=2124703 -OutFile WebView2Installer.exe
      - name: Install WebView2
        run: |
          # This downloads about 200 MB and takes about 5 minutes on my VM
          # So we could fault in WebView2 without the MSI if we needed.
          .\WebView2Installer.exe /install
      - name: Build assets
        run: |
          # Copied from `build.bat`, but I don't want to run `tauri build`
          pnpm install
          cp "node_modules/flowbite/dist/flowbite.min.js" "src/"
          pnpm tsc
          pnpm tailwindcss -i src/input.css -o src/output.css
      - name: Run smoke test
        run: cargo run -p firezone-windows-client -- smoke-test
```

Also I may be missing something, but is the "Smoke test failed" conditional necessary? GH Actions should fail the test if it doesn't exit successfully.

Reply:
When I went to split them up it looks like it stopped installing WebView2? The smoke test crashes with the "WebView not installed" error and it says it installed WebView in 5 seconds, that can't be right.
![image](https://github.com/firezone/firezone/assets/13400041/eddb1151-c712-4320-bc9c-1c76bd829e48)

Reply:
Got it fixed. 1 minute to install WebView2, then 5 minutes 30 seconds to compile the debug build. It completes faster than the release MSI build, since release builds are slow 
![image](https://github.com/firezone/firezone/assets/13400041/703a2c94-78a7-49fe-a8c1-322a57d688ce)

Reply:
Nice!